lib/ansible/modules/cloud/amazon/api_gateway.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '0.1'}
+
+DOCUMENTATION = '''
+---
+module: api-gateway
+short_description: Manage AWS API Gateway APIs
+description:
+     - Allows for the management of API Gatway APIs.
+version_added: '2.2'
+requirements: [ boto3 ]
+options:
+  api_id:
+    description:
+      - The ID of the API you want to manage
+    required: true
+  state:
+    description:
+      - NOT IMPLEMENTED Create or delete Lambda function
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  api_file:
```

Comment:
I think it'd be good to accept a couple ways of taking in API file. For example, users may want to use the `template` lookup like:

```
lambda:
  # make lambda functions
api_gateway:
  api_swagger: "{{ lookup('template', 'swagger_def.yml.j2') }}" 
  # ... other options ...
```

Since I imagine folks would want to define the API and define the functions in one playbook.